Objects/listobject.c

```diff
@@ -180,6 +180,22 @@ PyList_New(Py_ssize_t size)
     return (PyObject *) op;
 }
 
+static PyObject *
+list_new_prealloc(Py_ssize_t size)
+{
+    PyListObject *op = (PyListObject *) PyList_New(0);
+    if (size == 0 || op == NULL) {
+        return (PyObject *) op;
+    }
+    op->ob_item = PyMem_New(PyObject *, size);
```

Comment:
Please add a comment why you don't use calloc() here, with a reference to the bpo number.